README.md

```diff
@@ -2,4 +2,114 @@
+}
+```
+This code imports the Marine SDK and wraps the greeting function with the `#[marine]` macro. Every function wrapped in such way will be exported from a compiled Wasm module.
+
+To compile this code to Wasm, you need the Marine CLI tool. If you haven't installed the cli already, use `cargo install marine`:
+
+```bash
+marine build --release
+```
+
+Finally, you obtain a build manifest embedded into a Wasm binary and can interact with a module in the Marine REPL (mrepl). If you don't have it, install it with `cargo install mrepl`:
+
+```bash
+> marine info ./target/wasm32-wasi/release/greeting.wasm
+it version:  0.23.1
+sdk version: 0.6.0
+authors:     <user-name>
+version:     0.1.0
+description:
+repository:
+build time:  2022-05-15 14:52:37.865550 +00:00 UTC
+
+> mrepl --quiet
+1> load greeting ./target/wasm32-wasi/release/greeting.wasm
+module successfully loaded into App service
+elapsed time: 52.153308ms
+
+2> interface
+Application service interface:
+greeting:
+  fn greeting(name: String) -> String
+
+3> call greeting greeting "user"
+result: String("Hi, user")
+ elapsed time: 132.021µs
+```
+
+The complete guide of developing this simple module can be found [here](https://fluence.dev/docs/marine-book/quick-start/develop-a-single-module-service).
+
+
+## SDK components
+
+The SDK exports the following major components:
+- `#[marine]` procedure macro that can be applied to
+    - [function](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/export-functions) making it export from a module
+    - [external block](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/import-functions) making it imported from a module in a Marine-suitable way
+    - [structure](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/structures), making it usable as an argument of an export or an import function
+- [call parameters interface](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/call-parameters) intended to provide a set of module start parameters
+- [mounted binaries interface](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/mounted-binaries) that can be used to call a CLI tool
+- [module_manifest](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/module-manifest) macro intended to embed some info into a compiled Wasm
+- [logging stuff](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/logging) that allows adjusting a Marine logging mechanism
+
+
+## Supported Rust types
+
+At the moment, the `#[marine]` macro allows the following Rust types as an argument of export and import functions or the field of a structure:
+- one of the following Rust basic types: `bool`, `u8`, `u16`, `u32`, `u64`, `i8`, `i16`, `i32`, `i64`, `f32`, `f64`
+- strings `String`, `&str`
+- a vector of elements of the above types
+- a vector composed of vectors of the above type, where recursion is acceptable, e.g. the type `Vec<Vec<Vec<u8>>>` is permissible
```

Comment:
It is usually recommended to put "e.g." in two commas like this: ...is acceptable, e.g., the type...  But I doubt anyone ever minds this :-)